tools/debugging/extract_sp_exceptions.sh

```diff
@@ -42,94 +40,81 @@ else
-
-    for scenario_dir in "${scenarios_dir}"/*; do
-        scenario=$(basename "${scenario_dir}")
-        print_bold "${scenario}"
-        separator
-
-        for node_dir in $(find "${scenario_dir}" -maxdepth 1 -type d -name "node_*"); do
-            result=$(gunzip -c ${node_dir}/*.log.gz | jq --tab 'select (.error!=null or .exception!=null)')
-            if [[ $result == "" ]]; then
-                result=$(cat "${node_dir}"/*.stderr | grep -v Starting | grep -v Stopped)
-            fi
-            if [[ $result != "" ]]; then
-                print_bold "- Found error in ${node_dir}"
-                echo -e "${result}"
-                echo "${result}" > "${DESTINATION_DIR}/errors/${scenario}.node.log.gz"
-            fi
-        done
-
-        separator
-        sp_error=$(gunzip -c "${scenario_dir}"/scenario-player-run_*.log.gz | jq 'select (.error!=null or .exception!=null)')
-
-        if [[ $sp_error != "" ]]; then
-            print_bold "- SP reported an error in ${scenario_dir}"
-
-            echo -e "${sp_error}"
-            echo "${sp_error}" > "${DESTINATION_DIR}/errors/${scenario}.sp.log"
-        fi
 
-        separator
+    for scenario_logs in $(find "${SCENARIOS_DIR}" -maxdepth 2 -type f -name "scenario-player*.gz"); do
+        scenario_successful=$(gunzip -c  "$scenario_logs" | jq 'select(.result == "success")')
+
+        if [ -z "${scenario_successful}" ]; then
+            scenario=$(basename "${scenario_logs}")
+            scenario_dir=$(dirname "${scenario_logs}")
+
+            separator
+            print_bold "Scenario ${scenario} failed"
+            gunzip -c  "$scenario_logs" | jq 'select(.result != null) | .message'
+            separator
+
+            for node_dir in $(find "${scenario_dir}" -maxdepth 1 -type d -name "node_*"); do
+                gunzip -c ${node_dir}/*.log.gz | jq --tab 'select (.error!=null or .exception!=null)'
+                cat "${node_dir}"/*.stderr | grep -v Starting | grep -v Stopped
+            done
+
+            separator
+
+            gunzip -c "${scenario_dir}"/scenario-player-run_*.log.gz | jq 'select (.error!=null or .exception!=null)'
+        fi
     done;
 }
 
-# export the symbol to allow the subshell spawned by parallel to use it
-export -f download_pfs_logs
-export -f download_server_logs
+[ ! -d "$DESTINATION_DIR" ] && {
+    mkdir -p "${SCENARIOS_DIR}"
 
-print_bold "Downloading PFS logs"
+    print_bold "Downloading services logs"
+    download_service_logs ms-goerli-backup.gz ms-goerli.gz msrc-goerli-backup.gz msrc-goerli.gz pfs-goerli-with-fee.gz pfs-goerli.gz
+
+    print_bold "Downloading scenarios list"
+    download_server_logs $SCENARIO_REMOTE_URL_1
+    download_server_logs $SCENARIO_REMOTE_URL_2
+}
 
-download_pfs_logs pfs-goerli &
-download_pfs_logs pfs-goerli-with-fee &
-wait
+# search_for_failures expects the logs to be compressed
+find "${SCENARIOS_DIR}" -type f -iname '*.log' | xargs gzip -q
```

Comment:
this has to be fixed in the servers ,for some reason the logs are not always compressed